src/math/ColorManagement.js

```diff
@@ -0,0 +1,124 @@
+import { sRGBColorSpace, LinearSRGBColorSpace, DisplayP3ColorSpace, LinearACESCGColorSpace } from '../constants.js';
+import { Matrix3 } from './Matrix3.js';
+import { Vector3 } from './Vector3.js';
+
+export function SRGBToLinear( c ) {
+
+	return ( c < 0.04045 ) ? c * 0.0773993808 : Math.pow( c * 0.9478672986 + 0.0521327014, 2.4 );
+
+}
+
+export function LinearToSRGB( c ) {
+
+	return ( c < 0.0031308 ) ? c * 12.92 : 1.055 * ( Math.pow( c, 0.41666 ) ) - 0.055;
+
+}
+
+// JavaScript RGB-to-RGB transforms, defined as
+// FN[InputColorSpace][OutputColorSpace] callback functions.
+const FN = {
+	[ sRGBColorSpace ]: { [ LinearSRGBColorSpace ]: SRGBToLinear },
+	[ LinearSRGBColorSpace ]: { [ sRGBColorSpace ]: LinearToSRGB },
+};
+
+// Chromatic Adaptation Transform (CAT) RGB-to-RGB transforms, defined as
+// CAT02[InputColorSpace][OutputColorSpace] 3x3 matrices computed by
+// https://www.colour-science.org/apps/.
+const CAT02 = {
+	[ LinearACESCGColorSpace ]: {
+		[ sRGBColorSpace ]: [
+			1.7048873310, - 0.6241572745, - 0.0808867739,
+			- 0.1295209353, 1.1383993260, - 0.0087792418,
+			- 0.0241270599, - 0.1246206123, 1.1488221099,
+		],
+		[ DisplayP3ColorSpace ]: [
+			1.3793363837, - 0.3112868172, - 0.0680495665,
+			- 0.0687964722, 1.0799570656, - 0.0111605934,
+			- 0.0022666792, - 0.0417050150, 1.0439716942,
+		],
+	},
+	[ sRGBColorSpace ]: {
+		[ LinearACESCGColorSpace ]: [
+			0.6131178129, 0.3411819959, 0.0457873443,
+			0.0699340823, 0.9181030375, 0.0119327755,
+			0.0204629926, 0.1067686634, 0.8727159106,
+		],
+	},
+	[ DisplayP3ColorSpace ]: {
+		[ LinearACESCGColorSpace ]: [
+			0.7357429995, 0.2140109244, 0.0502460760,
+			0.0469048867, 0.9399887338, 0.0131063796,
+			0.0034712266, 0.0380157226, 0.9585130508,
+		],
+	},
+};
+
+const _matrix = /*@__PURE__*/new Matrix3();
+const _vector = /*@__PURE__*/new Vector3();
+
+export const ColorManagement = {
+
+	enabled: false,
```

Comment:
Saying that "Color Management" is disabled by setting `ColorManagement.enabled` equal to `false` doesn't entirely feel correct. In order for color to truly be "unmanaged" here textures need to not be transformed before being passed into the shader which is not how three.js currently works or works after this PR. I've mentioned before that what happens currently feels closer to operating in an sRGB color space but even that isn't a completely accurate depiction of what's currently happening. Three.js' current handling of color and textures is a bit of it's own thing currently. It seems best to refer to it as a "legacy" mode rather than trying to adopt the "Unmanaged Color" term and full deprecate it later. If we really want an option for unmanaged color in the future then more changes to textures will need to be made.

Reply:
I think I understand your concerns on this, but I am not sure how best to address it. If we think of this in terms of modes or workflows, there are several, and we may or may not want to support them all. Input colors and textures may come from various color spaces. In each mode ...

1. **Color Managed (NEW)**: Colors and textures are converted to the Linear-sRGB working color space during upload or processing.
2. **Non Color Managed**: Colors and textures are _not_ converted to the Linear-sRGB working color space automatically. This workflow does not exist in three.js today (textures are always converted if marked sRGB), and we may or may not want to add it.
3. **Current / "Legacy"**: Textures are converted to Linear-sRGB during upload; colors are not modified. This is the current behavior, and there are 2-3 likely reasons a user might choose it:
  - a. User is not familiar with color management, uses `outputEncoding=LinearEncoding`, and is effectively using a _gamma_ workflow. This is something I want to discourage and eventually deprecate.
  - b. User is comfortable with color management, uses `outputEncoding=sRGBEncoding`, understands color literals must be provided in Linear-sRGB, and is effectively using a _linear_ workflow.
  - c. User's project does not require lighting, and/or is compositing 2D layers with blending that does not require a linear ('scene referred') working color space. Any color space conversions (especially the additional precision required for internal Linear-sRGB render targets) is just unwanted performance overhead for this user, so they might prefer to do everything in a display-referred color space like sRGB. Sprite-based experiences like [Oat the Goat](http://oatthegoat.co.nz/) might be good examples.

I believe (3c) and (3b) are valid use cases that we should not forget. Perhaps we push these users toward (2) instead of (3), but I don't think we can expect them to be happy adopting (1). So our choice of naming (`ColorManagement.enabled` or otherwise)should be respectful of that.

Ideas:

- Push group (3b) toward option (2), deprecate (3a) and (3b), name the flag `ColorManagement.enabled`
- Don't support (2), name the flag something that invites (3b) and (3c) but discourages (3a) else, like:
    - `Color.enableTransform = false`
    - `ColorManagement.sourceColorSpace = LinearSRGBColorSpace`
    - `ColorManagement.workingColorSpace = sRGBColorSpace`
    - ...

Reply:
On further thought, users in group (3b) cannot use workflow (2), because their textures still need to be converted... I guess I'm tempted then to support (1) and (3) only for now, and to decide later whether (2) is necessary and whether (3) can be deprecated.

Reply:
My vote is to call the current behavior "Legacy" in some way and then in the long run implement a non color managed mode if we feel it's needed. At the least I just don't think we can use terminology that implies the current three.js behavior is the same as unmanaged color. But I do feel the "Legacy" mode should be deprecated and removed at some point.

I agree that user group "A" is should be discouraged in the long term. And I think user group "C" is really best suited by a true non color managed work flow. Regarding user group "B" --

> On further thought, users in group (3b) cannot use workflow (2), because their textures still need to be converted...

I don't think an unmanaged color workflow would be the easiest path for user group B, anyway. The only thing user group B would have to do to migrate to the new color managed mode is change any place that colors are explicitly set to use an sRGB hex _or_ explicitly pass that the color is in the Linear sRGB color space. It's a migration task but it's not a big one. Realistically I think this is the case that most users are going to fall in to given the current three.js recommendations.

And in terms of converting textures I don't think that's the case as long as the textures are already sRGB. With unmanaged color "texture.colorSpace" and "renderer.outputColorSpace" would be entirely ignored so the lighting calculations be off but the final displayed color would look reasonable as long as sRGB colors are passed into the Color constructor and textures are original stored in sRGB color space.

Reply:
> And I think user group "C" is really best suited by a true non color managed work flow...

I'm having second thoughts about (C) now. Consider CSS – it has nothing to do with PBR rendering, but CSS Color Module Level 4 adds support for Display P3 (wide-gamut) colors and actually _does_ need a color managed workflow. How do you feel about ... 

```
ColorManagement.mode = THREE.FullColorManagement;    // (a)
ColorManagement.mode = THREE.LegacyColorManagement;  // (b)
```

... with the caveats that "full" is more aspirational than truth at the moment, and "legacy" should eventually be removed.

Reply:
In the long term if Oat the Goat wanted to properly support wider-gamut displays then yes it would need color management enabled. In the short term (or long term if they limit their capability to the narrower sRGB color space) then disabling color management should work.

But overall yeah I agree I think a `ColorManagement.mode` field with a "Legacy" option should work in the short term and should be removed in the long term.

Reply:
I think I've settled a boolean `legacyMode` flag, if that sounds OK:

```js
ColorManagement.legacyMode = false; // default 'true'
```

The inverse of that could also work, e.g. `standardMode` defaulting to 'false'.